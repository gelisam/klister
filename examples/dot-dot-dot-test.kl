#lang "prelude.kl"

(import (rename (shift "prelude.kl" 1)
                [syntax-case raw-syntax-case]
                [unquote raw-unquote]
                [... raw-...]))
(import "dot-dot-dot-test-keywords.kl")
(import (shift "dot-dot-dot-test-keywords.kl" 1))
(import (shift "dot-dot-dot.kl" 1))

(define-macros
  ([my-macro
    (lambda (stx)
      (syntax-case stx (keyword)
        [(_ ((,a ,b) (,c ,d)))
         (pure `'(,a ,b ,c ,d))]
        [(_ (keyword ,tail ...))
         (pure `'(keyword-prefixed ,tail ... end-of-list))]
        [(_ (,e ...))
         (pure `'(ordinary-list ,e ... end-of-list))]))]))

(example (my-macro ((1 2) (3 4))))
(example (my-macro (keyword bar baz)))
(example (my-macro (foo bar baz)))
