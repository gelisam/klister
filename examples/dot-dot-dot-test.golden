#[dot-dot-dot-test.kl:17.17-17.31]<(1 2 3 4)> : Syntax
#[dot-dot-dot-test.kl:20.17-20.75]
 <(keyword-identifier-prefixed bar baz end-of-list)> : Syntax
#[dot-dot-dot-test.kl:22.17-22.58]<(keyword-prefixed 3 4 end-of-list)> : Syntax
#[dot-dot-dot-test.kl:24.17-24.52]<(ordinary-list foo bar baz end-of-list)> : Syntax
